relay-metrics/src/aggregator.rs

```diff
@@ -624,10 +625,12 @@ impl Aggregator {
                 let bucket_interval = self.config.bucket_interval;
                 let cost_tracker = &mut self.cost_tracker;
 
-                for (key, entry) in self
-                    .buckets
-                    .extract_if(|_, entry| force || entry.elapsed(now))
-                {
+                while let Some((_, entry)) = self.buckets.peek() {
+                    if !entry.elapsed(now) && !force {
+                        break;
+                    }
+
+                    let (key, entry) = self.buckets.pop().expect("pop after peek");
```

Comment:
This is awkward, right? For our own priority map I'm considering writing an interface that lets you do `let value = queue.peek().remove()` instead of `queue.pop()`.

Reply:
Yeah it is, I proposed an [API here](https://github.com/garro95/priority-queue/issues/55).

What could also be nice is also something like in [heed](https://docs.rs/heed/0.20.3/heed/struct.RwRevIter.html) where you have extra functionality on the iterator.